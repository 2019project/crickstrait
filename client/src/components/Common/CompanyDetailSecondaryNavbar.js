import React, { Component } from "react";
import {
  getCompanyDetailById,
  getVolatility
} from "../../actions/CompanyDetail";
import { connect } from "react-redux";
import companylogo from "./stockslogo.PNG";

// importing css file
import "../../styles/CompanyDetailSecondaryNavbar.css";
import { Link } from "react-router-dom";

export class CompanyDetailSecondaryNavbar extends Component {
  componentDidMount() {
    this.props.getVolatility("AAPL");
  }
  render() {
    return (
      <div>
        <div id="parent">
          {this.props.company ? (
            <div>
              {this.props.company.map(coms => (
                <>
                  <div id="company-details-tab-container">
                    <>
                      <div id="secondary-navbar-img-name-container">
                        <div id="secondary-navbar-img">
                          <img
                            id="stocks_img"
                            src={
                              coms.image == null
                                ? companylogo
                                : "data:image/jpeg;base64," + coms.image
                            }
                          />
                        </div>

                        <div id="secondary-navbar-ticker_name">
                          <p>{coms.ticker_name}</p>
                          <p id="CompanyDetailsIndexClose">
                            {coms.last_share_price}

                            <sub id="secondary-navbar-sub">USD</sub>
                          </p>
                          <p id="last_date">
                            CLOSED PRICE (
                            {new Date(coms.share_date).toLocaleDateString(
                              "en-IN",
                              {
                                month: "short",
                                day: "2-digit",
                                year: "numeric"
                              }
                            )}
                            )
                          </p>
                        </div>
                        <div />
                      </div>
                      <ul id="company-detail-content-container">
                        <Link
                          className="company-link"
                          // PASSING TO COMPANY DETAIL PAGE WITH THE ID WHICH IS MAPPED FROM THE REDUCER
                          to={{
                            pathname: "/companydetail/" + coms.id
                          }}
                        >
                          <span
                            id="company-detail-overview-click"
                            // onClick={() => {
                            //   // changing the values of the state while clicking, the clicked component will be set to true others will be false
                            //   // this.setState({
                            //   //   overview: true,
                            //   //   financial: false,
                            //   //   analysis: false,
                            //   // });
                            //   console.log("Overview clicked");
                            // }}
                          >
                            <li
                              id="comapany-detail-overview-li"
                              // different css properties based on the value of the state
                              className={
                                this.props.selected === "overview"
                                  ? "options-selected-li"
                                  : "options-li"
                              }
                            >
                              Overview
                            </li>
                          </span>
                        </Link>
                        {/* Financials  */}
                        {/* <div id="company-detail-financial"> */}

                        <div id="company-detail-financial">
                          <Link
                            className="company-link"
                            to={{
                              pathname: "/financial/" + coms.id
                            }}
                          >
                            <span
                              id="company-financial-click"
                              // onClick={() => {
                              //   this.setState({
                              //     overview: false,
                              //     financial: true,
                              //     analysis: false,
                              //   });
                              // }}
                            >
                              <li
                                id="company-detail-financial-li"
                                className={
                                  this.props.selected === "financial"
                                    ? "options-selected-li"
                                    : "options-li"
                                }
                              >
                                Financials
                              </li>
                            </span>
                          </Link>
                        </div>

                        <div id="company-detail-analysis">
                          <Link
                            className="company-link"
                            to={{
                              pathname: "/analysis/" + coms.id
                            }}
                          >
                            <span
                              id="company-analysis-click"
                              // onClick={() => {
                              //   this.setState({
                              //     overview: false,
                              //     financial: false,
                              //     analysis: true
                              //   });
                              // }}
                            >
                              <li
                                id="company-detail-analysis-li"
                                className={
                                  this.props.selected === "analysis"
                                    ? "options-selected-li"
                                    : "options-li"
                                }
                              >
                                Analysis
                              </li>
                            </span>
                          </Link>
                        </div>
                      </ul>
                      <div id="secondary-nav-values">
                        <div id="voltality">
                          <p> {this.props.voltality} </p>
                          <div id="voltality_tooltip">
                            <div id="kk">VOLATILITY </div>
                            <div id="abc">
                              <i class="fa fa-question-circle" />
                              <span id="volatility">
                                Volatility is a statistical measure of the
                                dispersion of returns for a given security or
                                market index. In most cases, the higher the
                                volatility, the riskier the security. Volatility
                                is often measured as either the standard
                                deviation or variance between returns from that
                                same security or market index.
                              </span>
                            </div>
                          </div>
                        </div>
                        {/* ------------------------------------------------------------------ */}
                        <div id="voltality">
                          <p id="market_cap"> {coms.last_market_cap} </p>
                          <div id="kk">MARKET CAP</div>
                        </div>
                        {/* --------------------------------------------------------------------------- */}
                      </div>
                    </>

                    <div id="bla">
                      {localStorage.getItem("token") ? (
                        <div id="secondary-navbar-bla">
                          {/* second grid of secondaru navbar */}
                          <button
                            id="downloadButton"
                            onClick={async () => {
                              fetch(
                                "http://localhost:2001/api/companydetail/indicatorsdata/" +
                                  coms.ticker_name
                              ).then(response => {
                                response.blob().then(blob => {
                                  let url = window.URL.createObjectURL(blob);
                                  let a = document.createElement("a");
                                  a.href = url;
                                  a.download = coms.ticker_name + ".csv";
                                  a.click();
                                });
                                //window.location.href = response.url;
                              });
                              // this.props.downloadOHLC("AAPL");
                            }}
                          >
                            <i class="fa fa-download" /> Download
                          </button>
                        </div>
                      ) : (
                        <div style={{ display: "none" }}></div>
                      )}
                    </div>
                  </div>
                  <></>
                </>
              ))}
            </div>
          ) : (
            <p />
          )}
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  company: state.CompanyDetailReducer.company,
  voltality: state.CompanyDetailReducer.voltality
});
export default connect(mapStateToProps, {
  getCompanyDetailById,
  getVolatility
})(CompanyDetailSecondaryNavbar);
