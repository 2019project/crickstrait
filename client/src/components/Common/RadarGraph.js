import React, { Component } from "react";
import { Polar } from "react-chartjs-2";
import { connect } from "react-redux";

export class RadarGraph extends Component {
  state = {
    marksData: {
      labels: [
        "Market Cap",
        "Net Profit",
        "Share Price",
        "Revenue",
        "Total Assets",
        "EBIT"
      ],
      datasets: [
        {
          fill: true,
          backgroundColor: [
            "#FF6384",
            "#4BC0C0",
            "#FFCE56",
            "#E7E9ED",
            "#36A2EB",
            "#2c3e50"
          ],

          data: this.props.data
        }
      ]
    }
  };
  render() {
    return (
      <div>
        <Polar
          options={{
            legend: false,
            labels: false,
            responsive: true
          }}
          data={this.state.marksData}
        />
      </div>
    );
  }
}
const mapStateToProps = state => ({});
export default connect(mapStateToProps, {})(RadarGraph);
