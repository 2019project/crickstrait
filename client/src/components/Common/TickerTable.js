import React, { Component } from "react";
import "../../styles/TickerTable.css";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

export class TickerTable extends Component {
  render() {
    console.log(this.props.isIndex);
    console.log(this.props.tableHeaders);
    return (
      <div className="table__container">
        <table className="ticker__table">
          <thead className="ticker__tableHeader">
            <tr>
              {this.props.tableHeaders.map((data, index) => (
                <th>{data}</th>
              ))}
            </tr>
          </thead>
          {console.log("tabledata", typeof this.props.tableData)}
          {this.props.tableData.map((current, index) => (
            <tr>
              {this.props.isIndex == true ? (
                <Link
                  to={{ pathname: "/indexProfile/" + current.ticker_id }}
                  style={{ textDecoration: "none" }}
                >
                  <td>{current.ticker_name ? current.ticker_name : null}</td>
                </Link>
              ) : (
                <Link
                  to={{ pathname: "/companydetail/" + current.ticker_id }}
                  style={{ textDecoration: "none" }}
                >
                  <td>{current.ticker_name ? current.ticker_name : null}</td>
                </Link>
              )}
              {current.tickerValues
                ? Object.keys(current.tickerValues).map((keyName, index) => (
                    <>
                      {current.tickerValues[keyName].substring(0, 1) == "-" ? (
                        <td style={{ color: "#ff4d4d" }}>
                          {current.tickerValues[keyName]}
                        </td>
                      ) : current.tickerValues[keyName].substring(0, 1) ==
                        "+" ? (
                        <td style={{ color: "#27ae60" }}>
                          {current.tickerValues[keyName]}
                        </td>
                      ) : (
                        <td>{current.tickerValues[keyName]}</td>
                      )}
                    </>
                  ))
                : null}
            </tr>
          ))}
        </table>
      </div>
    );
  }
}
const mapStateToProps = state => ({});
export default connect(mapStateToProps, {})(TickerTable);
