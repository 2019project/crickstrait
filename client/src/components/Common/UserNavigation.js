import React, { Component } from "react";
import "../../styles/UserNavigation.css";
import wallet from "./wallet.png";
import { Link } from "react-router-dom";
import jwt_decode from "jwt-decode";

export default class UserProfile extends Component {
  state = {
    buyStocks: true,
    myStocks: false,
    historyStocks: false
  };

  render() {
    var decode = jwt_decode(localStorage.getItem("token"));
    return (
      <div>
        <div id="userNavigationContainer">
          <div id="userNavigation">
            <h1>Hi Nikhil !</h1>
            <img src={wallet} id="wallet" />{" "}
            <span id="walletPrice">
              $
              {String(this.props.wallet).length != 0
                ? this.props.wallet
                : decode.wallet}
            </span>
            <div id="userNavigationButtonContainer">
              <Link to="/profile">
                <button
                  id={this.state.buyStocks ? "buyStocks-active" : "buyStocks"}
                  onClick={() =>
                    this.setState({
                      buyStocks: true,
                      myStocks: false,
                      historyStocks: false
                    })
                  }
                >
                  Buy Stocks
                </button>
              </Link>
              <Link to="/buyStocks">
                <button
                  id={this.state.myStocks ? "myStocks-active" : "myStocks"}
                  onClick={() =>
                    this.setState({
                      buyStocks: false,
                      myStocks: true,
                      historyStocks: false
                    })
                  }
                >
                  My Stocks
                </button>
              </Link>
              <Link to="/userhistory">
                <button
                  id={
                    this.state.historyStocks
                      ? "historyStocks-active"
                      : "historyStocks"
                  }
                  onClick={() =>
                    this.setState({
                      buyStocks: false,
                      myStocks: false,
                      historyStocks: true
                    })
                  }
                >
                  History
                </button>
              </Link>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
