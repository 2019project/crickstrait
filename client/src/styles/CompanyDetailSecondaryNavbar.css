/* FOR DESKTOP  */

#company-detail-content-container > div {
  color: #2c3e50;
  font-size: 20px;
}

/* parent container */
/* main grid  */
#parent {
  box-shadow: 0px 4px 6px 0px #cacaca;
}
#company-details-tab-container {
  padding: 0px 50px 0px 50px;
  /* grid-template-rows: 1fr 1fr; */
  display: grid;
  grid-template-areas: "overview  analysis";
  /* "chart chart"; */
  grid-template-columns: 1fr 1fr;
  /* overflow-x: scroll; */
  overflow-y: scroll;
  /* box-shadow: 0px 4px 6px 0px #cacaca; */
}
#bla {
  grid-area: analysis;
  margin: 0px;
  padding: 30px 0px 0px 0px;
}
#secondary-navbar-bla {
  margin: 0px;
  padding: 0px;
}

#abc #volatility {
  visibility: hidden;
  width: 168px;
  background-color: #2c3e50;
  color: #fff;
  text-align: justify;
  border-radius: 6px;
  padding: 15px 15px 15px 15px;
  font-size: 14px;
  font-weight: 400;

  /* Position the tooltip */
  position: absolute;
  z-index: 1;
}

#fa-question-circle {
  margin: 10px 10px 10px 5px;
}
#abc:hover #volatility {
  visibility: visible;
}

#abc {
  margin: 0px;
  padding: 0px;
}
/* child within parent container */
/* grid within grid  */

/* tabs */
#company-detail-content-container {
  padding: 20px 0px 0px 0px;
  grid-template-rows: 1fr;
  display: flex;
  grid-template-areas: "a b c";
  /* "chart chart"; */
  grid-template-columns: 1fr 1fr 1fr;
  /* overflow-x: scroll; */
  overflow-y: scroll;
  grid-gap: 15px;
  list-style-type: none;
  font-size: 20px;
}

#comapany-detail-overview-li {
  grid-area: a;
  margin: 0px 50px 0px 0px;
}
#company-detail-financial-li {
  grid-area: b;
  margin: 0px 50px 0px 0px;
}

#secondary-download-button {
  grid-area: c;
}

.options-selected-li {
  font-weight: 600;
  cursor: pointer;
  /* padding-bottom: 7px; */
  color: #2980b9;
  border-bottom: 3px solid #2980b9;
}
.options-li {
  color: #272727;

  text-align: center;
}
.company-link {
  text-decoration: none;
}

/* for image and ticker name grid  */

#secondary-navbar-img-name-container {
  display: flex;
}

#secondary-navbar-ticker_name {
  margin: 18px 10px 10px 10px;
  font-size: 25px;
  color: #2c3e50;
  font-weight: bold;
}
#secondary-navbar-img {
  margin: 30px 10px 10px 10px;
}
#last_date {
  font-size: 12px;
  color: #707070;
}
#indexClose {
  float: left;
  text-align: center;
  font-size: 17px;
  font-weight: bold;
}
#secondary-navbar-sub {
  font-size: 10px;
}
#kk {
  font-size: 12px;
  color: #707070;
  margin: 0px;
  padding: 0px;
}

#voltality_tooltip {
  display: flex;
  margin: 0px;
  padding: 0px;
}

#voltality {
  margin: 0px;
  padding: 0px 0px 0px 60px;
}

#voltality p {
  font-size: 17px;
  font-weight: 600;
}

#secondary-nav-values {
  display: flex;
  justify-content: flex-end;
}

/* FOR MOBILE  */
@media only screen and (max-width: 426px) {
  #company-details-tab-container {
    padding: 0px 20px 0px 20px;
    /* grid-template-rows: 2fr; */
    display: grid;
    grid-template-areas:
      "overview"
      " financial";
    grid-template-columns: 1fr;
    /* overflow-x: scroll; */
    overflow-y: scroll;
    box-shadow: 0px 4px 6px 0px #cacaca;
  }

  #secondary-nav-values {
    grid-area: 2/4;
    grid-column: 1/4;
  }
  #secondary-navbar-img-name-container {
    grid-area: 1/4;
    grid-column: 1/4;
    justify-content: center;
  }
  #bla {
    grid-area: 3/4;
    grid-column: 1/4;
  }

  #secondary-nav-values {
    justify-content: center;
  }

  #voltality {
    margin: 0px;
    padding: 0px 10px 0px 30px;
  }

  #bla {
    padding: 0px 70px 0px 0px;
  }

  #comapany-detail-overview-li {
    grid-area: a;
    margin: 0px 12px 0px 0px;
  }

  #company-detail-financial-li {
    grid-area: b;
    margin: 0px 12px 0px 0px;
  }
}
