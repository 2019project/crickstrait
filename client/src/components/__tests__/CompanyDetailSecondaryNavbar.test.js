import React from "react";
import { shallow, mount } from "enzyme";
import { CompanyDetailSecondaryNavbar } from "../Common/CompanyDetailSecondaryNavbar";

const getCompanyDetailById = jest.fn();
const getOhlcChart = jest.fn();
const getGaugeCompany1 = jest.fn();
const click = jest.fn();

const company = [
  {
    last_share_price: "aapl",
    share_date: "123456",
    last_market_cap: "123456",
    // industry: "Computer",
    // employees: "132",
    // profile: "profile",
  },
];

const wrapper = shallow(
  <CompanyDetailSecondaryNavbar
    getCompanyDetailById={getCompanyDetailById}
    company={company}
    click={click}
  />
);

describe("Testing Company Detail Secondary navbar Component", () => {
  //   afterEach(() => {
  //     jest.clearAllMocks()

  it("should mount the component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should display the market cap", () => {
    expect(wrapper.find("#indexClose").text()).toBe("aaplUSD");
  });

  it("should display the market cap", () => {
    expect(wrapper.find("#market_cap").text()).toBe(" 123456 ");
  });

  //   download button
  it("checks for on edit button function to be called", () => {
    const e = { preventDefault: () => {} };
    jest.spyOn(e, "preventDefault");
    wrapper.find("#downloadButton").simulate("click", e);
  });

  //overview
  it("checks for on edit button function to be called", () => {
    const e = { preventDefault: () => {} };
    jest.spyOn(e, "preventDefault");
    wrapper.find("#company-detail-overview-click").simulate("click", e);
  });

  //financial
  it("checks for on edit button function to be called", () => {
    const e = { preventDefault: () => {} };
    jest.spyOn(e, "preventDefault");
    wrapper.find("#company-financial-click").simulate("click", e);
  });

  //analysis
  it("checks for on edit button function to be called", () => {
    const e = { preventDefault: () => {} };
    jest.spyOn(e, "preventDefault");
    wrapper.find("#company-detail-analysis").simulate("click", e);
  });

  it("should display the market cap", () => {
    expect(wrapper.find("#volatility").text()).toBe(
      "Volatility is a statistical measure of the dispersion of returns for a given security or market index. In most cases, the higher the volatility, the riskier the security. Volatility is often measured as either the standard deviation or variance between returns from that same security or market index."
    );
  });
  it("should display the market cap", () => {
    expect(wrapper.find("#kk").text()).toBe("VOLATILITY");
  });

  //on click function
});
