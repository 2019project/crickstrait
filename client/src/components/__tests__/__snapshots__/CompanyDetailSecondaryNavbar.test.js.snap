// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Company Detail Secondary navbar Component should mount the component 1`] = `
<div>
  <div
    id="parent"
  >
    <div>
      <div
        id="company-details-tab-container"
      >
        <div
          id="secondary-navbar-img-name-container"
        >
          <div
            id="secondary-navbar-img"
          >
            <img
              id="stocks_img"
              src="stockslogo.PNG"
            />
          </div>
          <div
            id="secondary-navbar-ticker_name"
          >
            <p />
            <p
              id="indexClose"
            >
              aapl
              <sub
                id="secondary-navbar-sub"
              >
                USD
              </sub>
            </p>
            <p
              id="last_date"
            >
              CLOSE (
              Jan 01, 123456
              )
            </p>
          </div>
          <div />
        </div>
        <ul
          id="company-detail-content-container"
        >
          <Link
            className="company-link"
            to={
              Object {
                "pathname": "/companydetail/undefined",
              }
            }
          >
            <span
              id="company-detail-overview-click"
              onClick={[Function]}
            >
              <li
                className="options-selected-li"
                id="comapany-detail-overview-li"
              >
                Overview
              </li>
            </span>
          </Link>
          <div
            id="company-detail-financial"
          >
            <Link
              className="company-link"
              to={
                Object {
                  "pathname": "/financial/undefined",
                }
              }
            >
              <span
                id="company-financial-click"
                onClick={[Function]}
              >
                <li
                  className="options-li"
                  id="company-detail-financial-li"
                >
                  Financials
                </li>
              </span>
            </Link>
          </div>
          <div
            id="company-detail-analysis"
          >
            <Link
              className="company-link"
              to={
                Object {
                  "pathname": "/analysis/undefined",
                }
              }
            >
              <span
                id="company-analysis-click"
                onClick={[Function]}
              >
                <li
                  className="options-li"
                  id="company-detail-analysis-li"
                >
                  Analysis
                </li>
              </span>
            </Link>
          </div>
        </ul>
        <div
          id="secondary-nav-values"
        >
          <div
            id="voltality"
          >
            <p>
               123456 
            </p>
            <div
              id="voltality_tooltip"
            >
              <div
                id="kk"
              >
                VOLATILITY 
              </div>
              <div
                id="abc"
              >
                <i
                  class="fa fa-question-circle"
                />
                <span
                  id="volatility"
                >
                  Volatility is a statistical measure of the dispersion of returns for a given security or market index. In most cases, the higher the volatility, the riskier the security. Volatility is often measured as either the standard deviation or variance between returns from that same security or market index.
                </span>
              </div>
            </div>
          </div>
          <div
            id="voltality"
          >
            <p
              id="market_cap"
            >
               
              123456
               
            </p>
            <div
              id="kk"
            >
              MARKET CAP
            </div>
          </div>
        </div>
        <div
          id="bla"
        >
          <div
            id="secondary-navbar-bla"
          >
            <button
              id="downloadButton"
              onClick={[Function]}
            >
              <i
                class="fa fa-download"
              />
               Download
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
